### ISSUE TITLE
Simple MQTT wilcards don't work

### ISSUE THREAD

**Issue by vladoro** [mqttjs/MQTT.js#962]

**Comment by vladoro** [comment]
mqtt.js: version: 3.0.0
mosquitto server version 1.6.2

## MQTTBox client (mac)
1. I subscribe to 'cast/iot/font/#'.
2. I publish the message with topic cast/iot/font/test
3. I receive the message.

## mqtt.js no wildcard
1. I subscribe to 'cast/iot/font/test'.
2.  I publish the message with topic cast/iot/font/test
3. **I receive the message**

## mqtt.js with wildcard
1. I subscribe to 'cast/iot/font/#' or ('cast/iot/font/+')
2.  I publish the message with topic cast/iot/font/test
3. **I don't receive the message**

## Code source
```
const mqtt = require('mqtt');
const client = mqtt.connect(
  process.env.MQTT_URL,
  {
    username: MQTT_USERNAME,
    password: MQTT_PASSWORD,
    port: MQTT_PORT,
    clientId: MQTT_CLIENTID,
  }
);
client.on('connect', () => client.subscribe('cast/iot/font/#'));
client.on('message', (topic, message) => console.debug(topic, message));
```

**Comment by kazazes** [comment]
I can reproduce this. Running `mosquitto_sub` with the same parameters, subscribed to `#` confirms that there are messages being missed. Also, subscriptions without wildcards work, confirming it's not a connection issue.

**Comment by thomasphorton** [comment]
I'm seeing the same thing- if I use the example in the documentation, modified with a few console logs:

```
var mqtt = require('mqtt')
var client  = mqtt.connect('mqtt://localhost:1883')

const topic = `#`;

client.on('connect', function () {
  client.subscribe(topic, function (err) {
    if (err) {
      console.log(err);
    }
    else {
      console.log(topic);
      client.publish(topic, 'Hello mqtt')
    }
  })
})

client.on('message', function (topic, message) {
  // message is Buffer
  console.log(message.toString())
  client.end()
})
```

I get a constant stream of
```
#
#
#
#
```

**Comment by thomasphorton** [comment]
Turns out writing a comment on an issue is all it takes for things to start clicking into place.

I took my mosquitto broker off of daemon mode and ran it in the window so I could see the logs- I'm getting a client error/disconnect, so it looks like it's just reconnecting, subscribing, failing over and over again and maybe not an issue with the wildcard. It is odd that this doesn't happen if I pass a 'normal' topic string though.
